### PR TITLE
Afk V2

### LINKED ISSUE #65: request: afk v2 - automatically turn off afk when message sent
Many people want afk to turn off automatically when user types a message. Maybe made a separate version for that.

### PR DESCRIPTION
**Pull request by DaviiD1337**
Resolves #65

### CODE REVIEW

File: afk/afk.go.tmpl [hunk 1]
```diff
@@ -5,41 +5,68 @@
...
-		{{ .User.Mention }}, I removed your AFK.
-	{{ else }}
-		Please either provide an afk message with `-afk <reason>`, with a duration `-afk -d <duration> <reason>` or you can remove your AFK with `-afk` (you either did not provide a message or you tried to turn off AFK when you were not).
-	{{ end }}
+
+{{/* CONFIGURATION VALUE STARTS HERE */}}
+{{ $removeAfkOnMessage := false }}{{/* Remove the AFK database from an user if he/she types something in chat. */}}
+{{/* CONFIGURATION VALUE ENDS HERE */}}
+
+{{/* Dont edit below if you don't know what you're doing ! */}}
```
Comment (Jo3-L) [inline afk/afk.go.tmpl:17]: superfluous comment, not editing code is implied as the configuration variables end above

File: afk/afk.go.tmpl [hunk 2]
```diff
@@ -5,41 +5,68 @@
...
+{{ $time := 0 }}{{ $afk := dbGet .User.ID "afk" }}
+{{ if .Args }}
+    {{ $args := .Args }}
```
Comment (Jo3-L) [inline afk/afk.go.tmpl:20]:
```suggestion
{{ if $args := .Args }}
```

File: afk/afk.go.tmpl [hunk 3]
```diff
@@ -5,41 +5,68 @@
...
+                    {{- $skip = false }}
+                    {{- $message = joinStr " " $message $v }}
+                {{- else if $skip }}
+                    {{- $skip = false }}
+                {{- end -}}
+            {{ end }}
+            {{ $parsedDur := 0 }}
+            {{ with and $duration (toDuration $duration) }} {{ $parsedDur = . }} {{ end }}
+            {{ if $parsedDur }}
+                {{ dbSetExpire .User.ID "afk" $message (div $parsedDur 1000000000) }}
```
Comment (Jo3-L) [inline afk/afk.go.tmpl:46]:
```suggestion
                {{ dbSetExpire .User.ID "afk" $message (toInt $parsedDur.Seconds) }}
```

**Review by Jo3-L — CHANGES_REQUESTED** [review]

File: afk/afk.go.tmpl [hunk 4]
```diff
@@ -5,41 +5,66 @@
...
-		Please either provide an afk message with `-afk <reason>`, with a duration `-afk -d <duration> <reason>` or you can remove your AFK with `-afk` (you either did not provide a message or you tried to turn off AFK when you were not).
-	{{ end }}
+
+{{/* CONFIGURATION VALUE STARTS HERE */}}
+{{ $removeAfkOnMessage := false }}{{/* Remove the AFK database from an user if he/she types something in chat. */}}
+{{/* CONFIGURATION VALUE ENDS HERE */}}
+
+{{ $time := 0 }}{{ $afk := dbGet .User.ID "afk" }}
+{{ if $args := .Args }}
+    {{ if eq (lower (index $args 0)) "?afk" }}
```
Comment (Jo3-L) [inline afk/afk.go.tmpl:19]: Would prefer that this handle mention prefix too, but not a blocker.

File: afk/afk.go.tmpl [hunk 5]
```diff
@@ -5,41 +5,66 @@
...
+            Welcome back, {{ .User.Mention }}, I removed your AFK.
+            {{ dbDel .User.ID "afk" }}
+        {{ else }}
+            Please either provide an afk message with `-afk <reason>`, with a duration `-afk -d <duration> <reason>` or you can remove your AFK with `-afk` (you either did not provide a message or you tried to turn off AFK when you were not).
+        {{ end }}
+    {{ else if .Message.Mentions }}
+        {{ with (dbGet (index .Message.Mentions 0).ID "afk") }}
+            {{ $user := userArg .UserID }}
+            {{ $eta := "" }}
+            {{ if gt .ExpiresAt.Unix 0 }} {{ $eta = humanizeDurationSeconds (.ExpiresAt.Sub currentTime) | printf "*%s will be back in around %s.*" $user.Username }} {{ end }}
```
Comment (Jo3-L) [inline afk/afk.go.tmpl:57]:
```suggestion
            {{ if not .ExpiresAt.IsZero }} {{ $eta = humanizeDurationSeconds (.ExpiresAt.Sub currentTime) | printf "*%s will be back in around %s.*" $user.Username }} {{ end }}
```

File: afk/afk.go.tmpl (hunk 4 above)
Comment (DaviiD1337) [reply afk/afk.go.tmpl:19]: YES TY LOOKS DMS <3
TAHTS WHAT I FOUND THE BUG <3

**Review by Jo3-L — APPROVED** [review]

**Comment by Jo3-L** [comment]
Thanks, and Merry Christmas! ☃️

**Comment by DaviiiD1337** [comment]
Ty, indeed!

**Pull request merged**